client/src/components/Endpoints/getIcon.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import { Plugin, GPTIcon, AnthropicIcon } from '~/components/svg';
+import { useAuthContext } from '~/hooks';
+import { cn } from '~/utils';
+import { IconProps } from '~/common';
+
+const getIcon: React.FC<IconProps> = (props) => {
+  const { size = 30, isCreatedByUser, button, model = true, endpoint, error, jailbreak } = props;
+
+  // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
One last thing, I just noticed this eslint disable line. This and the minimalist function are already structured like React functional components so we should just make them. Rename it to Icon.tsx and the function as well, remove the comment and treat it like a regular React.FC(which it even has a type for).